fms/models/__init__.py

```diff
@@ -65,6 +66,47 @@ def list_variants(architecture: str):
     return list(__models[architecture].keys())
 
 
+def __maybe_infer_model_variant(
+    architecture: str,
+    variant: str,
+    model_path: Optional[str],
+    source: Optional[str],
+    **kwargs,
+) -> Tuple[str, str, Optional[str], Optional[str], Dict[str, Any]]:
+    """Infer the model variant configuration from different sources, currently only supported sources are hf"""
+    extra_kwargs = kwargs
+
+    if architecture in ("hf_pretrained", "hf_configured"):
+        logging.info(f"inferring model configuration from {variant}")
+
+        if len(kwargs) > 0:
```

Comment:
can we accept changes to the kwargs when we are doing "hf_configured"? I understand why we don't want to accept them on "hf_pretrained"

Reply:
> can we accept changes to the kwargs when we are doing "hf_configured"? I understand why we don't want to accept them on "hf_pretrained"

Yes good point, we could modify if we are doing "hf_configured". Will update with that change

Reply:
Updated in latest commit